pkg/infra/baseagent.go

```diff
@@ -142,7 +145,7 @@ func (a *Base) TakeFood(amountOfFood food.FoodType) food.FoodType {
 		a.Log("An agent has taken food", Fields{"floor": a.floor, "amount": foodTaken})
 		return foodTaken
 	}
-	return 0
+	return -1
```

Comment:
Not sure if this is a good suggestion or not, but might be worth returning an error as well as a quantity?

This would let us differentiate between the cases?

```suggestion
	return 0, errors.New("agent has already eaten/platform isn't on level")
```

Reply:
I think -1 is a simple way of signaling that we didn´t eat (instead of 0 which could indicate that we ate 0) As @udai-arneja commented below, I think it's not worth making it more complicated and -1, 0, other x would be easy to handle